scripts/compare-fidelity-results.js

```diff
@@ -0,0 +1,144 @@
+
+for (let i = 0; i < candidateThresholds.length; ++i) {
+  const candidateThreshold = candidateThresholds[i];
+
+  if (remainingGoldenThresholds.length === 0) {
+    warn(`${goldenResultsDirectory} tests ${
+        candidateThresholds.length -
+        goldenThresholds.length} fewer thresholds than ${
+        candidateResultsRectory}`);
+    exit();
+  }
+
+  const goldenThreshold = remainingGoldenThresholds.shift();
+
+  if (candidateThreshold !== goldenThreshold) {
+    warn(`Candidate uses threshold ${
+        candidateThreshold} where golden uses threshold ${goldenThreshold}`);
+    exit();
+  }
+}
+
+for (const goldenScenario of goldenScenarios) {
+  goldenScenarioMap.set(goldenScenario.slug, goldenScenario);
+}
+
+for (const candidateScenario of candidateScenarios) {
+  const {slug} = candidateScenario;
+
+  if (!goldenScenarioMap.has(slug)) {
+    warn(`${goldenResultsDirectory} does not include scenario "${
+        slug}" found in ${candidateResultsDirectory}`);
+    continue;
+  }
+
+  const goldenScenario = goldenScenarioMap.get(slug);
+  const candidateAnalysis = require(path.resolve(
+      path.join(candidateResultsDirectory, slug, 'analysis.json')));
+  const goldenAnalysis = require(
+      path.resolve(path.join(goldenResultsDirectory, slug, 'analysis.json')));
+
+  const goldenGoldenMap = new Map();
+
+  for (const goldenGolden of goldenScenario.goldens) {
+    goldenGoldenMap.set(goldenGolden.name, goldenGolden);
+  }
+
+  for (let i = 0; i < candidateScenario.goldens.length; ++i) {
+    const candidateGolden = candidateScenario.goldens[i];
+    if (!goldenGoldenMap.has(candidateGolden.name)) {
+      warn(`${goldenResultsDirectory} does not include an analysis of "${
+          candidateGolden.name}" for scenario "${slug}" found in ${
+          candidateResultsDirectory}`);
+      continue;
+    }
+
+    const goldenGolden = goldenGoldenMap.get(candidateGolden.name);
+
+    const candidateResults = candidateAnalysis.analysisResults[i];
+    const goldenResults = goldenAnalysis.analysisResults[i];
+
+    for (let j = 0; j < candidateThresholds.length; ++j) {
+      const threshold = candidateThresholds[j];
+
+      const candidateThresholdResult = candidateResults[j];
+      const goldenThresholdResult = goldenResults[j];
+
+      for (const key in candidateThresholdResult) {
+        if (!(key in goldenThresholdResult)) {
+          warn(`Golden analysis of "${candidateGolden.name}" in scenario "${
+              slug}" is missing metric "${key}"`);
+          continue;
+        }
+
+        const delta =
+            goldenThresholdResult[key] - candidateThresholdResult[key];
+
+        const comparisonDescription =
+            `<model-viewer> <-> ${candidateGolden.name}`;
+        const comparisonConstraints =
+            `"${slug}/${key}" @ threshold ${threshold}`;
+        const percentage = `${(delta * 100).toFixed(2)}%`;
+
+        if (delta > ALERT_THRESHOLD) {
```

Comment:
No, because it's only bad if the delta is positive. Unless I flipped things around somewhere by accident... either way, the intended behavior is sign-dependent.

Reply:
Oh, right.. a decrease is good?

I might restructure so the code is (very bad pseudo-code and wording):

```
if (delta > ALERT) alert
else if (delta > 0) warn('is more different');
else "great! the two sources are less different"
```

Reply:
👍 WIll fix

Reply:
> Oh, right.. a decrease is good?

Negative deltas = more similarity than before